ruby/lib/minitest/queue.rb

```diff
@@ -256,6 +259,25 @@ def run_from_queue(reporter, *)
           reporter.record(result)
         end
       end
+
+      queue.monitor.shutdown!
+    end
+
+    def with_heartbeat(example)
+      finished = false
+      t = Thread.start do
+        heartbeats = queue.config.timeout.to_f * 1_000 # convert to milliseconds
+        while heartbeats > 0 && !finished
+          queue.monitor.tick!(example.id)
+          heartbeats -= 1
+          sleep 0.001
```

Comment:
I suspect you did this for testing, but ticking every 1ms is a lot, IMO you can tick every 1 or 2 seconds just fine.